reference.module

```diff
@@ -302,26 +324,55 @@ function reference_menu() {
 }
 
 /**
- * Menu callback for providing autocomplete values to the reference fields.
+ * Provides values for the reference field using taxonomy terms.
  */
```

Comment:
@moonray Can you give me an overview of why we need a function specific to taxonomy terms? We are trying to avoid any entity-specific functions in this module, other than as temporary shims until we've updated core to ensure core entities provide all the entity api functionality we need.